### ISSUE TITLE
Edit message composer has superfluous escaping

### ISSUE THREAD

**Issue by germain-gg** [element-hq/element-web#22456]
### Steps to reproduce

1. Send `matrix-react-sdk/res/themes/light/css/_light.scss` as  message in Element
2. Hit the up arrow to edit the message

#### What did you expect?

To see the edit message composer appear and prefilled with `matrix-react-sdk/res/themes/light/css/_light.scss` 

#### What happened instead?

<img width="448" alt="Screen Shot 2022-06-06 at 09 05 50" src="https://user-images.githubusercontent.com/769871/172164867-802dd30e-d58d-41d1-a9e8-6da61387dd85.png">
<img width="402" alt="Screen Shot 2022-06-06 at 13 56 28" src="https://user-images.githubusercontent.com/769871/172164925-f40a301f-f2c0-4d6c-8fc9-dcdc377c215d.png">

Notice the incorrect escaping before the underscore

### Will you send logs?

No

**Comment by gsouquet** [comment]
I disagree with the triaging here. The escaping is incorrect. My message does not contain those backslash and to a non technical user this will look out of place

It could even make them move the cursor to delete those characters which is a big ask

**Comment by t3chguy** [comment]
@gsouquet that's a whole class of bugs

https://github.com/vector-im/element-web/issues/22342
https://github.com/vector-im/element-web/issues/20387
https://github.com/vector-im/element-web/issues/10808

**Comment by anoadragon453** [comment]
Also chiming in with my own instance of this while editing a message. The editing composer showed an extra backslash, whereas the original message only contained one:

![image](https://user-images.githubusercontent.com/1342360/183912190-ae020c36-a5ad-43aa-930d-9ee8cf314dbd.png)

<details>

<summary>Event source</summary>

```
{
  "type": "m.room.message",
  "content": {
    "org.matrix.msc1767.message": [
      {
        "body": "\\",
        "mimetype": "text/plain"
      },
      {
        "body": "\\",
        "mimetype": "text/html"
      }
    ],
    "body": "\\",
    "msgtype": "m.text",
    "format": "org.matrix.custom.html",
    "formatted_body": "\\"
  }
}
```

</details>

Looking at the event source, I'm confused why `body` has two backslashes...

**Comment by t3chguy** [comment]
> Looking at the event source, I'm confused why body has two backslashes...

Otherwise it'd be escaping the `"`

**Comment by t3chguy** [comment]
Yeah so that is just the re-escaping code being naive and not checking if the escaping is necessary

**Comment by babolivier** [comment]
Worth noting it affects more than just what the sender sees in the composer, as it can also affect code blocks, as seen in one of the supporter rooms earlier this week:

![image](https://user-images.githubusercontent.com/5547783/185370722-db9f3f7f-2978-42b2-94ed-2e30ff8c18b9.png)

(since escaping backlashes aren't automagically hidden in code blocks)

**Comment by robintown** [comment]
@babolivier Please open a separate issue for that, so we can keep this one focused on the benign cases of unnecessary escaping. Incidentally, I can't reproduce that issue, so please provide repro steps if you can.

**Comment by StyXman** [comment]
So this was closed but I still see it in https://app.element.io/ is it just a matter of the code not being deployed yet?

**Comment by t3chguy** [comment]
@StyXman this issue hasn't been closed though?

![image](https://user-images.githubusercontent.com/2403652/191688102-28b7003b-fd54-479c-bd6b-8a425e282e2c.png)

**Comment by babolivier** [comment]
> Please open a separate issue for that, so we can keep this one focused on the benign cases of unnecessary escaping. Incidentally, I can't reproduce that issue, so please provide repro steps if you can.

Sorry I completely missed this comment. iirc I was reporting an issue with a message that @anoadragon453 sent. I cannot seem to be able to repro it either. So I'll let Andrew open a new issue if it happens again.

**Comment by anoadragon453** [comment]
I'm not able to reproduce the issue in [your comment](https://github.com/vector-im/element-web/issues/22456#issuecomment-1219305641) at this point @babolivier.

I can still reproduce the one in the message editing composer [as shown here](https://github.com/vector-im/element-web/issues/22456#issuecomment-1210669323) however.

Element Desktop Nightly 2022090801, Arch Linux.

**Comment by StyXman** [comment]
> @StyXman this issue hasn't been closed though?

yeah, sorry, I got confused by this:

![image](https://user-images.githubusercontent.com/167327/197381141-59f1c6ce-8bda-4359-acbf-6b8d65ec1043.png)
